pages/pipelines/environment_variables.md.erb

```diff
@@ -392,6 +374,36 @@ The secret access key for your S3 IAM user, for use with [private S3 buckets](/d
 
 Example: `"wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"`
 
+<h3 class="h3-caps">BUILDKITE_SHELL</h3>
+
+The value of the `shell` [agent configuration option](/docs/agent/v3/configuration). The value cannot be modified as an environment variable.
+
+Example: `"/bin/bash -e -c"`
+
+<h3 class="h3-caps">BUILDKITE_SOURCE</h3>
+
+The source of the event that created the build. The value cannot be modified.
+
+Values: `"webhook"`, `"api"`, `"ui"`, `"trigger_job"`, or `"schedule"`
+
+<h3 class="h3-caps">BUILDKITE_SSH_KEYSCAN</h3>
+
+The opposite of the value of the `no-ssh-keyscan` [agent configuration option](/docs/agent/v3/configuration). The value cannot be modified as an environment variable.
+
+Values: `"true"` and `"false"`
+
+<h3 class="h3-caps">BUILDKITE_STEP_ID</h3>
+
+A unique string that identifies a step. This value cannot be modified.
+
+Example: "080b7d73-986d-4a39-a510-b34f9faf4710"
+
+<h3 class="h3-caps">BUILDKITE_STEP_KEY</h3>
+
+The value of the `key` [command step option](/docs/pipelines/command-step#command-step-attributes), a unique string set by you to identify a step. This value cannot be modified as an environment variable. 
```

Comment:
`s/option/attribute/`?

Reply:
Ah, I see we use option above too! 🤔 Whatever really!

Reply:
I like attribute better for this one, given that the link to the command step docs has 'attribute' in it 👍🏻